routes/api.py

```diff
@@ -480,6 +481,65 @@ def query_history_api(version=1):
     return better_jsonify(valid=True)
 
 
+_SPEECH_API_KEY = None
+_SPEECH_API_KEY_PATH = os.path.join(
+    os.path.dirname(__file__), "..", "resources", "GreynirSpeechKey.txt"
+)
+
+
+def _speech_api_key():
+    """ Load speech key from file. """
+    global _SPEECH_API_KEY
+    if _SPEECH_API_KEY is None:
+        with open(_SPEECH_API_KEY_PATH) as f:
+            _SPEECH_API_KEY = f.read().strip()
+    return _SPEECH_API_KEY
+
+
+@routes.route("/speech.api", methods=["GET", "POST"])
+@routes.route("/speech.api/v<int:version>", methods=["GET", "POST"])
+def speech_api(version=1):
+    """ Send in text, receive URL to speech-synthesised audio file. """
+
+    if not (1 <= version <= 1):
+        return better_jsonify(valid=False, reason="Unsupported version")
+
+    reply = dict(err=True)
+
+    text = request.values.get("text")
+    if not text:
+        return better_jsonify(**reply)
+
+    key = request.values.get("key")
+    fmt = request.values.get("format", "ssml")
+    if fmt not in ["text", "ssml"]:
+        fmt = "ssml"
+    voice_id = request.values.get("voice_id")
+    speed = request.values.get("voice_id", "Dora")
```

Comment:
Something mixed up here - should probably be `speed = request.values.get("speed", 1.0)`